force-app/main/default/classes/ServiceService.cls

```diff
@@ -34,13 +75,18 @@ public with sharing class ServiceService {
                 contactId
             )
         ) {
-            programIds.add(engagement.Program__c);
-            result.get('engagements')
-                .add(convertObjectToOption(engagement, engageProgram));
+            if (activeProgramEngagementStages.contains(engagement.Stage__c)) {
+                programIds.add(engagement.Program__c);
+                result.get('engagements')
```

Comment:
This predates this PR but noting that we prolly should at least have these as final variables at the top of the method.